### PR TITLE
Support for NSQ 1.0.0

### PR DESCRIPTION
**Pull request by netantho**
NSQ 1.0.0 was recently released https://github.com/nsqio/nsq/releases/tag/v1.0.0-compat

This pull request:
- adds a travis building matrix for NSQ versions 0.3.5 to 1.0.0, use --node-id flag for nsqd 1.0.0 which replaces --worker-id
- replaces deprecated POST /put by POST /pub in the tests

Building matrix: https://travis-ci.org/netantho/elixir_nsq/builds/215295750

### CODE REVIEW

File: test/consumer_test.exs [hunk 1]
```diff
@@ -49,10 +49,10 @@ defmodule NSQ.ConsumerTest do
     NSQ.Consumer.event_manager(consumer)
       |> GenEvent.add_handler(NSQ.ConsumerTest.EventForwarder, self())
 
-    HTTP.post("http://127.0.0.1:6751/put?topic=#{@test_topic}", [body: "hello"])
+    HTTP.post("http://127.0.0.1:6751/pub?topic=#{@test_topic}", [body: "hello"])
```
Comment (MaxPower15) [inline test/consumer_test.exs:52]: So I suppose it works fine from the tests passing, but just to confirm: /pub is supported on older versions of nsqd, right?

**Comment by MaxPower15** [comment]
Nice change! Left one question for you, but it'll be nice to automatically test against a bunch of different versions.

**Comment by netantho** [comment]
Maybe I'm missing something but I don't see your question.
Did you forget to submit your review by any chance?

**Comment by MaxPower15** [comment]
Whoops, so I did! It should be there now I think. Sorry about that.

File: test/consumer_test.exs (hunk 1 above)
Comment (mreiferson) [reply test/consumer_test.exs:52]: yes 😁,  I think all the way back to 0.2.29

Comment (netantho) [reply test/consumer_test.exs:52]: /pub came with the V1 spec according to the changelog https://github.com/nsqio/nsq/blob/c22751c7e63cacb28e61c13b84efe95beb192910/ChangeLog.md#100-compat---2017-03-21
From what I can find in github, that was introduced on Jul 19, 2013
https://github.com/nsqio/nsq/commit/8b50fe86cdf1872282f1e3e80ade405c4d27d68a#diff-17367c04222666c193afbedd39b5fff1R26
and came with NSQ 0.2.2
https://github.com/nsqio/nsq/blob/master/ChangeLog.md#0222---2013-08-26

Comment (MaxPower15) [reply test/consumer_test.exs:52]: Great! That's plenty backwards compatibility as far as I'm concerned.

**Pull request merged**